### PR TITLE
feat(frontend): add differential loading for web apps

### PR DESCRIPTION
**Pull request by FrozenPandaz**
_[Please make sure you have read the submission guidelines before posting an PR](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#-submitting-pr)_

## Current Behavior (This is the behavior we have today, before the PR is merged)

The web builder does not utilize differential loading:

## Expected Behavior (This is the new behavior we can expect after the PR is merged)

The web builder utilizes differential loading.

It is enabled by default.

2 sets of bundles are produced: 1 for evergreen browsers (ones that support es2015) and 1 for older browsers.

### CODE REVIEW

File: packages/web/src/builders/build/build.impl.ts [hunk 1]
```diff
@@ -36,40 +48,98 @@ export function run(
...
-            );
-          }
-
           context.logger.info(stats.toString());
         }
-      })
-    )
+      };
+      return from(configs).pipe(
+        mergeScan(
```
Comment (vsavkin) [inline packages/web/src/builders/build/build.impl.ts]: would not forkJin work better here? I might be misunderstanding what you are trying to do

File: packages/web/src/utils/web.config.ts [hunk 2]
```diff
@@ -1,46 +1,72 @@
...
 import { LoggerApi } from '@angular-devkit/core/src/logger';
-import { resolve } from 'path';
-import typescript = require('typescript');
+import { join, resolve } from 'path';
 import { WebBuildBuilderOptions } from '../builders/build/build.impl';
 import { convertBuildOptions } from './normalize';
 import { readTsConfig } from '@nrwl/workspace';
 import { getBaseWebpackPartial } from './config';
+import { IndexHtmlWebpackPlugin } from '@angular-devkit/build-angular/src/angular-cli-files/plugins/index-html-webpack-plugin';
+import { ScriptTarget } from '@schematics/angular/third_party/github.com/Microsoft/TypeScript/lib/typescript';
```
Comment (vsavkin) [inline packages/web/src/utils/web.config.ts]: we import it from 'typescript' in another file. why?

**Review by vsavkin — CHANGES_REQUESTED** [review]
I'm a bit concerned that we depend on angular-devkit/build-angular here. How hard would it be not to?

**Comment by Jordan-Hall** [comment]
@mbinic are you able to make this change or do you mind if I jump on the PR and make the change required?

File: packages/web/src/builders/build/build.impl.spec.ts [hunk 3]
```diff
@@ -4,19 +4,22 @@ import { of } from 'rxjs';
 import * as buildWebpack from '@angular-devkit/build-webpack';
 jest.mock('tsconfig-paths-webpack-plugin');
 import TsConfigPathsPlugin from 'tsconfig-paths-webpack-plugin';
+import * as webConfigUtils from '../../utils/web.config';
 import { getMockContext, MockBuilderContext } from '../../utils/testing';
 import { join } from 'path';
-import * as fs from 'fs';
+import * as indexHtmlUtils from '@angular-devkit/build-angular/src/angular-cli-files/utilities/index-file/write-index-html';
```
[the remark this answers is not in the document]
Comment (FrozenPandaz) [reply packages/web/src/builders/build/build.impl.spec.ts]: There is a way. But for time, can we address this in a future release/PR?

**Pull request merged**